azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -1263,6 +1338,29 @@ private void cleanOlderProjectVersion(Connection connection, int projectId,
     }
   }
 
+  private static class ProjectIdsByEventTypeResultHandler implements
+          ResultSetHandler<Set<Integer>> {
+    private static final String SELECT_PROJECTS_BY_EVENT_TYPE =
+            "SELECT distinct(project_id) from project_events WHERE `event_type`=?";
```

Comment:
This will be costly as it will scan significant data. Should we keep this in the main sync block ?